Userland/Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -1234,8 +1235,76 @@ static NonnullRefPtr<StyleValue const> interpolate_box_shadow(DOM::Element& elem
 
 static NonnullRefPtr<StyleValue const> interpolate_value(DOM::Element& element, StyleValue const& from, StyleValue const& to, float delta)
 {
-    if (from.type() != to.type())
+    if (from.type() != to.type()) {
+        // Handle mixed percentage and dimension types
+
+        struct NumericBaseTypeAndDefault {
+            CSSNumericType::BaseType base_type;
+            ValueComparingNonnullRefPtr<CSS::StyleValue> default_value;
+        };
+        static constexpr auto numeric_base_type_and_default = [](StyleValue const& value) -> Optional<NumericBaseTypeAndDefault> {
+            switch (value.type()) {
+            case StyleValue::Type::Length: {
+                static auto default_length_value = LengthStyleValue::create(Length::make_px(0));
+                return NumericBaseTypeAndDefault { CSSNumericType::BaseType::Length, default_length_value };
+            }
+            case StyleValue::Type::Angle: {
+                static auto default_angle_value = AngleStyleValue::create(Angle::make_degrees(0));
+                return NumericBaseTypeAndDefault { CSSNumericType::BaseType::Angle, default_angle_value };
+            }
+            case StyleValue::Type::Time: {
+                static auto default_time_value = TimeStyleValue::create(Time::make_seconds(0));
+                return NumericBaseTypeAndDefault { CSSNumericType::BaseType::Time, default_time_value };
+            }
+            case StyleValue::Type::Frequency: {
+                static auto default_frequency_value = FrequencyStyleValue::create(Frequency::make_hertz(0));
+                return NumericBaseTypeAndDefault { CSSNumericType::BaseType::Frequency, default_frequency_value };
+            }
+            case StyleValue::Type::Percentage: {
+                static auto default_percentage_value = PercentageStyleValue::create(Percentage { 0.0 });
+                return NumericBaseTypeAndDefault { CSSNumericType::BaseType::Percent, default_percentage_value };
+            }
+            default:
+                return {};
+            }
+        };
+
+        static constexpr auto to_calculation_node = [](StyleValue const& value) -> NonnullOwnPtr<CalculationNode> {
+            if (value.is_percentage())
+                return NumericCalculationNode::create(value.as_percentage().percentage());
+            if (value.is_length())
+                return NumericCalculationNode::create(value.as_length().length());
+            if (value.is_angle())
+                return NumericCalculationNode::create(value.as_angle().angle());
+            if (value.is_frequency())
+                return NumericCalculationNode::create(value.as_frequency().frequency());
+            if (value.is_time())
+                return NumericCalculationNode::create(value.as_time().time());
```

Comment:
Nit: These are in a *different* arbitrary order. Is it based on how common the types are? That's alright.

Reply:
Yes, but there's no reason this can't also be a switch statement in the same (now sorted) order as the one above!